test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public class AspirationalExamples
+    {
+        public class IrisPrediction
+        {
+            public string PredictedLabel;
+        }
+
+        public class IrisExample
+        {
+            public float[] Features;
+        }
+
+        public void FirstExperienceWithML()
+        {
+            // This is the 'getting started with ML' example, how we see it in our new API.
+            // It currently doesn't compile, let alone work, but we still can discuss and improve the syntax.
+
+            // Initialize the environment.
+            using (var env = new TlcEnvironment())
+            {
+                // Load the data into the system.
+                string dataPath = "iris-data.txt";
+                var data = TextReader.FitAndRead(env, dataPath, c => (Label: c.LoadString(0), Features: c.LoadFloat(1, 4)));
```

Comment:
>Features:

Do we have to use the term `Features` or could you break it into SepalLength, SepalWidth, PetalLength, PetalWidth?

Reply:
Either is doable. Do you think breaking it down is preferable? I made it into 'Features' because I wanted to demonstrate that you don't HAVE to bring in features as scalars.

Reply:
The nice thing about breaking it apart is in your inference code.

You don't have to say 

```C#
engine.Predict(new IrisExample { Features = new[] { 3.3f, 1.6f, 0.2f, 5.1f } })
```

Which one was `SepalLength`, and which one was `SepalWidth`?  Oops - I accidentally put Petal properties first in the array, and Sepal properties 2nd.

Where if you have them broken apart you can say:

```C#
engine.Predict(new IrisExample 
{
    SepalWidth = 3.3f, 
    SepalLength = 1.6f, 
    PetalWidth = 0.2f,
    PetalLength = 5.1f
 });
```

Which makes for much more readable and maintainable code IMO.

Reply:
That is true when you have 4 features, but I observed that a more common case is to have a dozen to a hundred. In this case, demonstrating that we CAN handle this with a vector column without explicitly naming them all is valuable. But I will rework the example to have separate scalars for features.

---
In reply to: [207697383](https://github.com/dotnet/machinelearning/pull/639#discussion_r207697383) [](ancestors = 207697383)